### ISSUE TITLE
false positive: assignment should start at column x when using preprocessors

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#787]
I am aware that the rule is beta, so just giving this feedback on when using preprocess conditional code:
````
def var cc1 as char no-undo.
def var cc2 as char no-undo.

&if false &then
   // nothing
&else
   if true then 
      cc1 = "foo".
   else
      cc1 = "bar".
&endif
cc2 = "something else".
````
The rule flags that cc2 should be starting at column 4 (instead of the 1 it is currently at).

Maybe just checking that the indent has been reduced, instead of demanding an exact position, could make the rule more usable.

But, just ran into another case:

````
def var cc as char no-undo.

if false then 
   assign
      cc = "foo"
      .
&if true &then
   &if true &then
      cc = "bar".
   &endif
&endif
````
Which makes it all a bit very tricky.

**Comment by gquerret** [comment]
Right, preprocessor makes things very difficult. I personally indent the &IF &ELSE, but not the content of those blocks, so first block would be:
```
def var cc1 as char no-undo.
def var cc2 as char no-undo.

&if false &then
// nothing
&else
if true then 
  cc1 = "foo".
else
  cc1 = "bar".
&endif
cc2 = "something else".
```

And one space instead of two for nested &IF:
```
&if true &then
 &if true &then
if true then
  if true then
    xx = true. 
 &endif
&endif
```

I'm not 100% sure we have enough tokens from preprocessor during rules execution (some are discarded) to know exactly where the token should start.

**Comment by stefandrissen** [comment]
We use three spaces for any indentation, regular or preprocessor. I long time ago I experimented with having preprocessor indentation separate from code indentation, ie preprocessor indentation always starts at column 1. While this made sense in some cases - big advantage is that the actual code is aligned normally - in other cases it became very unclear what was going on. I'll see if I can make some time to revisit this.

**Comment by stefandrissen** [comment]
Had a quick bash at some reformatting, keeping actual code aligned with a single character outdent for &if / &else / &then / &endif works fine for me.

The only minor problem I have is that I cannot outdent if I am already at column 1, for example for procedure scoped variables - since they are the minority they will have to survive without an indent.
